src/main/java/com/microsoft/graph/authentication/TokenCredentialAuthProvider.java

```diff
@@ -0,0 +1,134 @@
+package com.microsoft.graph.authentication;
+
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import com.microsoft.graph.exceptions.AuthenticationException;
+import com.microsoft.graph.exceptions.Error;
+import com.microsoft.graph.exceptions.ErrorConstants;
+import com.microsoft.graph.httpcore.IHttpRequest;
+import okhttp3.Request;
+import com.microsoft.graph.exceptions.ErrorConstants.*;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+
+import javax.annotation.Nonnull;
+
+public class TokenCredentialAuthProvider implements ICoreAuthenticationProvider , IAuthenticationProvider {
+
+    //TokenCredential expected from user
+    private TokenCredential tokenCredential;
+    //Context options which can be optionally set by the user
+    private TokenRequestContext context;
+    // maximum delay to wait for token obtention
+    private Duration tokenBlockTimeout;
+
+    
+    //TODO: Upon further review from Peter, this should include a Null check in the case that this is the first request
+    //There should be just one constructor as scopes should probably not be optional
```

Comment:
Scopes is not an optional parameter. getToken() requires it.
Would `public TokenCredentialAuthProvider(@Nonnull final TokenCredential tokenCredential)` be useful if scopes are not being passed? 
Also, please rephrase this comment especially the part 'scopes should probably not be optional'.

Reply:
I reviewed the order of the parameters to make the overload constructors a bit more useful for optional parameters.
Also added default values, more defensive programing and removed the comments. Let me know what you think about the new version!